src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/InterceptedProjectProperties/AbstractBuildEventValueProvider.AbstractBuildEventHelper.cs

```diff
@@ -0,0 +1,192 @@
+                var execTask = projectXml.Targets
+                                    .Where(target =>
+                                        StringComparer.OrdinalIgnoreCase.Compare(GetTarget(target), BuildEvent) == 0 &&
+                                        StringComparer.OrdinalIgnoreCase.Compare(target.Name, TargetName) == 0)
+                                    .SelectMany(target => target.Tasks)
+                                    .Where(task => StringComparer.OrdinalIgnoreCase.Compare(task.Name, ExecTask) == 0)
+                                    .FirstOrDefault();
+                return (success: execTask != null, execTask: execTask);
+            }
+
+            private (bool success, ProjectTargetElement target) FindTargetToRemove(ProjectRootElement projectXml)
+            {
+                var foundTarget = projectXml.Targets
+                                        .Where(target =>
+                                            StringComparer.OrdinalIgnoreCase.Compare(GetTarget(target), BuildEvent) == 0 &&
+                                            target.Children.Count == 1 &&
+                                            target.Tasks.Count == 1 &&
+                                            StringComparer.OrdinalIgnoreCase.Compare(target.Tasks.First().Name, ExecTask) == 0)
+                                        .FirstOrDefault();
+                return (success: foundTarget != null, target: foundTarget);
+            }
+
+            private void SetParameter(ProjectRootElement projectXml, string unevaluatedPropertyValue)
+            {
+                var result = FindExecTaskInTargets(projectXml);
+
+                if (result.success == true)
+                {
+                    SetExecParameter(result.execTask, unevaluatedPropertyValue);
+                }
+                else
+                {
+                    var targetName = GetTargetName(projectXml);
+                    var target = projectXml.AddTarget(targetName);
+                    SetTargetDependencies(target);
+                    var execTask = target.AddTask(ExecTask);
+                    SetExecParameter(execTask, unevaluatedPropertyValue);
+                }
+            }
+
+            private void SetExecParameter(ProjectTaskElement execTask, string unevaluatedPropertyValue)
+                => execTask.SetParameter(Command, unevaluatedPropertyValue);
+
+            private string GetTargetName(ProjectRootElement projectXml)
+            {
+                var targetNames = new HashSet<string>(projectXml.Targets.Select(t => t.Name), StringComparer.OrdinalIgnoreCase);
+                var targetName = TargetName;
+                if (targetNames.Contains(targetName))
+                {
+                    targetName = FindNonCollidingName(targetName, targetNames);
+                }
+
+                return targetName;
+
+            }
+
+            private string FindNonCollidingName(string buildEvent, HashSet<string> targetNames)
```

Comment:
Let`s not do this - if the exact name doesn't match then that should be an error - this just complicates things without much value.